### DISCUSSION TITLE
Getting [Vue Router warn]: Finding ancestor route "/messages" failed for "/messages" after upgrading to 4.3.3

### DISCUSSION THREAD

**Discussion by martsyn** [vuejs/router#2282] [Help and Questions] [answered]
After the upgrade to 4.3.3 with the switch to binary search for insertMatcher ([#2137](https://github.com/vuejs/router/issues/2137)) by @skirtles-code I am getting three of messages like this in dev console:

    [Vue Router warn]: Finding ancestor route "/messages" failed for "/messages"

I have a big project with a lot of routes (plenty of nested routes and such) and having trouble replicating this in a smaller project. If I remove some random routes before the ones that show up in warnings, these warnings still show up with other routes. This and the fact that this warning has a comment `// This should never happen` before it in new function `findInsertionIndex` baffles me.

Would be happy if someone helps me out here or points me in the right direction to replicate this issue so I can understand what's wrong with my route setup or file a proper bug report. Unfortunately, sharing my project in its existing state is not very feasible.

**Comment by skirtles-code** [comment, marked as answer]
Could it be the same problem as #2266? That was caused by having two routes with the same `name`, with one being a descendant of the other.

There's a pending PR, #2267, that is intended to help identify this problem.

**Comment by martsyn** [comment]
This indeed was the case of name duplicated both on parent and child. Thank you!
